### PR TITLE
verkle: remove allocations for storage slot tree calculation

### PR DESCRIPTION
**Pull request by jsign**
This PR eliminates the usage of `big.Int` when calculating the tree-hash for a storage slot.
It does some careful calculations using `u256.Int` to avoid any potential overflow.

### CODE REVIEW

File: trie/utils/verkle.go [hunk 1]
```diff
@@ -279,22 +275,27 @@ func GetTreeKeyStorageSlotWithEvaluatedAddress(evaluated *verkle.Point, storageK
...
-	// Because the modulus is 256, it's the last byte of treeIndex
-	posBytes := pos.Bytes()
-	subIndex := posBytes[len(posBytes)-1]
+	// If the storage slot is in the main storage, we need to add the main storage offset.
 
-	return treeIndex, subIndex
+	// We first divide by VerkleNodeWidth to create room to avoid an overflow next.
+	pos.Rsh(&pos, uint(VerkleNodeWidthLog2))
+	// We add mainStorageOffset/VerkleNodeWidth which can't overflow.
+	pos.Add(&pos, mainStorageOffsetLshVerkleNodeWidth)
```
Comment (jsign) [inline trie/utils/verkle.go:296]: I was quite explicit in the comments, but adding a note here.

The TL;DR for this case is that we need to do `(MAIN_STORAGE_OFFSET + storage_key)/VERKLE_NODE_WIDTH`
The detail is that we avoid doing `MAIN_STORAGE_OFFSET + storage_key`, since that would overflow in a `uint256`; so we do it in parts.

First we do `storage_key/VERKLE_NODE_WIDTH`, and then we add `mainStorageOffsetLshVerkleNodeWidth` which is a constant literally `MAIN_STORAGE_OFFSET/VERKLE_NODE_WIDTH`.

Note doing this in part is correct since `MAIN_STORAGE_OFFSET` is a multiple of `VERKLE_NODE_WIDTH`, we we aren't "loosing precision" in the VERKLE_NODE_WIDTH lower bits having "carries".

This is similar to doing something like: `(100 + 12) / 10` which should be `11`. In this case it's correct doing `100/10+12/10` since also yields `11`. 

But note that if we had to do something like `(109+12)/10` it yields 12. But if we try doing `109/10+12/10` that is _incorrect_ since it yields `11`!

File: trie/utils/verkle.go [hunk 2]
```diff
@@ -279,22 +275,27 @@ func GetTreeKeyStorageSlotWithEvaluatedAddress(evaluated *verkle.Point, storageK
 }
 
 func GetTreeKeyStorageSlotTreeIndexes(storageKey []byte) (*uint256.Int, byte) {
```
Comment (gballet) [inline trie/utils/verkle.go:277]: That function no longer exists after the shapella rebase/your new access witness, would it make sense to try to adapt `GetTreeKeyWithEvaluatedAddess` with this technique?

**Review by gballet — COMMENTED** [review]
Looks like this PR no longer applies as this function no longer exists after merging the new access witness.

File: trie/utils/verkle.go (hunk 2 above)
Comment (jsign) [reply trie/utils/verkle.go:277]: Uhm, not sure I follow. `GetTreeKeyStorageSlotTreeIndexes` is used by `GetTreeKeyStorageSlotWithEvaluatedAddress` (L272 in this file; prev function); and this last function is used every time we need to get the key for a storage slot in the trie.

**Review by gballet — APPROVED** [review]
that was a rebase error, I tried and it works, showing some slight speed improvement (and less allocations)
